File: sumDigits.py
def sum_digits(n):
		"""Calculates the sum of the digits n
		>>> sum_digits(8)
		8
		>>> sum_digits(18)
		9
		>>> sum_digits(2018)
		11
		"""
		if n < 10:
			return n
		else:
			all_but_last, last = n // 10, n % 10
			return sum_digits(all_but_last) + last

def main():
	num = sum_digits(3)
	print(num)

if __name__ == '__main__':main()

File: sumDigits.java
public class sumDigits {

    public static void main(String[] args) {
        System.out.println(sum_digits(3));
    }

    private static int sum_digits(int n) {
        if (n < 10)
            return n;
        else {
            int all_but_last = (int) Math.floor((double) n / 10), last = n % 10;
            return (int) (sum_digits(all_but_last) + last);
        }
    }
}
